### ISSUE TITLE
Improve handling of mumble + system tray

### ISSUE THREAD

**Issue by murlakatamenka** [mumble-voip/mumble#3879]
1. Add option to start mumble minimized to tray like `mumble --tray`
Good for autostarting, invoking from shell, running as a cron job etc.

2. Remove `Server` -> `Hide mumble` menu item?
Seems not really useful because you can either minimize the window (1 click) or minimize it right to tray (1 click, if option "Hide in tray when minimized" is ticked). 1 click < 2 clicks, better UX, faster!

3. Allow closing to tray, not just minimizing

- Some apps are meant to be used mostly in background and invoked only when needed. At least such scenarios are possible and likely. Apps like `mumble`, `qBittorrent` (torrent client), `OBS Studio` (streaming client). I think it's nice for them to handle `close window` signal as a way to get rid of unnecessary window at the moment, not actually close. Or at least to provide such option to configure by the user. Example from `qBittorrent`:

![image](https://user-images.githubusercontent.com/7361274/68074409-bf2ad100-fdab-11e9-8263-66eae6a2c939.png)

- `mumble` already has a safety net from accidental closing ("Ask on quit while connected"), so being able to close to tray should work even better!

- This should be extremely useful for people who use tiling window managers. We don't even have these minimize, close buttons :smile: . We just close them with a hotkey. `qBittorrent` closes to tray and I'm happy. `mumble` doesn't, I have to use `Server `-> `Hide mumble` to minimize it to tray :laughing:  

4. Allow binding "Hide mumble" to a shortcut

Not sure about this, but this can serve as a temporary workaround for the items above. I'll look into source code, it should be not hard to implement since the functionality already exists.

Hope the feedback is useful.

---

Best regards.

**Comment by Krzmbrzl** [comment]
As far as I can tell the `Hide Mumble` option was introduced for #2737

**Comment by Tha14** [comment]
> 1. Add option to start mumble minimized to tray like `mumble --tray`
>        Good for autostarting, invoking from shell, running as a cron job etc.

I'd love to have the above added. It's quite annoying to have to minimize mumble on every boot.

**Comment by rejedai** [comment]
Has there been a solution to this problem yet? Maybe there is an option in the config?

**Comment by Hartmnt** [comment]
I think #6260 added a shortcut to hide/show Mumble. As for the other things: Over the years there have been quite a few bug reports regarding the system tray implementation. Just search for "tray" in the issue list.

The consensus seems to be that the tray icon implementation needs a complete rewrite. But this has not been a priority for the Mumble core devs yet.

I think it is safe to assume that such a change wont make it into 1.5 stable so at best this would land in 1.6 (let's hope that will take less time to iron out compared to 1.5)

**Comment by rejedai** [comment]
I've been using tray for a fair amount of time and haven't seen any problems, is this more of a windows issue or does it on linux too?

I use KDE Wayland.

**Comment by Hartmnt** [comment]
> I've been using tray for a fair amount of time and haven't seen any problems, is this more of a windows issue or does it on linux too?
> 
> I use KDE Wayland.

No, this is also a Linux issue. The best way to reproduce buggy tray behavior is to:

1) Make Mumble fullscreen
2) Click on the Mumble tray icon

Do this at your own risk, I have heard from people that might crash your X11 session in some configurations.

**Comment by rejedai** [comment]
It's definitely fine on wayland. I tried both maximize and press to tray icon; alse kwin fullscreen and alt+f4 (with enabled minimize to tray).
![Screenshot_2024_04_06_13_44_15](https://github.com/mumble-voip/mumble/assets/92713267/5e62586c-c2f9-4371-a46a-8b98379bee77)
